### PR TITLE
added callout about insecure use of env var

### LINKED ISSUE #226: Document security best practices for signing into the registry with Notation
As part of the release of Notation 1.0.0 we need to document the security best practices to sign into a registry using Notation. We should document the recommended way to sign in as well as other sign in options and what are the benefits and risks with using those. Things like using ENV variables or credentials as part of the CLI switches should be discouraged and the associated risks documented.

### PR DESCRIPTION
**Pull request by zr-msft**
fixes #226

### CODE REVIEW

**Review by FeynmanZhou — APPROVED** [review]
LGTM. This is an important callout for security best practice.

**Review by FeynmanZhou — APPROVED** [review]
LGTM. This is an important callout for security best practice.

**Comment by zr-msft** [comment]
moving it back to draft for now to add recommended auth approaches

**Comment by zr-msft** [comment]
Updated with additional callouts around auth methods

**Review by yizha1 — APPROVED** [review]
LGTM

File: content/en/docs/how-to/registry-authentication.md [hunk 1]
```diff
@@ -87,6 +71,28 @@ Use `notation login` to authenticate to an OCI-compliant registry.
...
+Set the `NOTATION_USERNAME` and `NOTATION_PASSWORD` environment variables to authenticate to an OCI-compliant registry.
+
+```bash
+export NOTATION_USERNAME="YOUR_REGISTRY_USERNAME"
+export NOTATION_PASSWORD="YOUR_REGISTRY_PASSWORD"
+```
+
+After the environment variables are set, you can use Notation with your registry without `notation login`.
+
+For security reasons, unset the environment variables after you log out from the registry. For example:
```
Comment (shizhMSFT) [inline content/en/docs/how-to/registry-authentication.md:89]: Users may be confused with the term `log out`. They may think they need to run some command to `log out`.

Comment (zr-msft) [reply content/en/docs/how-to/registry-authentication.md]: updated based on your feedback

**Review by toddysm — APPROVED** [review]

File: content/en/docs/how-to/registry-authentication.md [hunk 2]
```diff
@@ -87,6 +71,28 @@ Use `notation login` to authenticate to an OCI-compliant registry.
...
+Set the `NOTATION_USERNAME` and `NOTATION_PASSWORD` environment variables to authenticate to an OCI-compliant registry.
+
+```bash
+export NOTATION_USERNAME="YOUR_REGISTRY_USERNAME"
+export NOTATION_PASSWORD="YOUR_REGISTRY_PASSWORD"
+```
+
+After the environment variables are set, you can use Notation with your registry without `notation login`.
+
+For security reasons, unset the environment variables after you use `notation logout` to log out from the registry. For example:
```
[the remark this answers is not in the document]
Comment (zr-msft) [reply content/en/docs/how-to/registry-authentication.md]: you're right. I misread the section. 

Updated to make it a bit clearer

**Review by byronchien — APPROVED** [review]

File: content/en/docs/how-to/registry-authentication.md [hunk 3]
```diff
@@ -87,6 +71,28 @@ Use `notation login` to authenticate to an OCI-compliant registry.
...
 
+## Configure environment variables to authenticate to an OCI-compliant registry
+
+{{% alert title="Warning" color="warning" %}}
+Environment variables are not secure and should not be used in a production environment for storing sensitive data such as usernames and passwords.
+{{% /alert %}}
+
+Set the `NOTATION_USERNAME` and `NOTATION_PASSWORD` environment variables to authenticate to an OCI-compliant registry.
+
+```bash
```
Comment (priteshbandi) [inline content/en/docs/how-to/registry-authentication.md:82]:
```suggestion
```console
```

**Review by priteshbandi — APPROVED** [review]
LGTM

**Review by yizha1 — APPROVED** [review]
LGTM

**Pull request merged**